### PR TITLE
feature: add new dynamic table paramaters

### PR DESCRIPTION
**Pull request by HenkvanDyk**
## What are you changing in this pull request and why?

Documentation for features added here https://github.com/dbt-labs/dbt-snowflake/pull/893
## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/static/_redirects`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 1]
```diff
@@ -421,6 +425,7 @@ models:
     [on_configuration_change](/reference/resource-configs/on_configuration_change)="apply" | "continue" | "fail",
     [target_lag](#target-lag)="downstream" | "<integer> seconds | minutes | hours | days",
     [snowflake_warehouse](#configuring-virtual-warehouses)="<warehouse-name>",
+    [refresh_mode]
```
Comment (matthewshaver) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
    [refresh_mode](#refresh-mode)="auto",
```

**Comment by matthewshaver** [comment]
@HenkvanDyk and @dataders what is the status of this draft? Should we move forward here?

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 2]
```diff
@@ -360,6 +360,10 @@ Dynamic tables are supported with the following configuration parameters:
 | [`on_configuration_change`](/reference/resource-configs/on_configuration_change) | `<string>` | no       | `apply` | n/a                       |
 | [`target_lag`](#target-lag)                                                      | `<string>` | yes      |         | alter                     |
 | [`snowflake_warehouse`](#configuring-virtual-warehouses)                         | `<string>` | yes      |         | alter                     |
+| [`refresh_mode`](#refresh-mode)                                | `<string>` | no       |         | n/a                       |
+| [`initialize`](#initialize)                                  | `<string>` | no       |         | n/a                       |
+| [`comment`](#comment)                                     | `<string>` | no       |         | n/a                       |
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/snowflake-configs.md:365]: Per https://github.com/dbt-labs/dbt-snowflake/pull/1081 we removed `comment`
```suggestion
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 3]
```diff
@@ -440,6 +445,34 @@ Snowflake allows two configuration scenarios for scheduling automatic refreshes:
...
+Specifies the behavior of the initial refresh of the dynamic table
+
+  - ON_CREATE
+
+    Refreshes the dynamic table synchronously at creation.
+  - ON_SCHEDULE
+
+    Refreshes the dynamic table at the next scheduled refresh.
+
+### Comment
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/snowflake-configs.md:473]: per https://github.com/dbt-labs/dbt-snowflake/pull/1081 removing `comment`
```suggestion
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 4]
```diff
@@ -440,6 +445,34 @@ Snowflake allows two configuration scenarios for scheduling automatic refreshes:
...
+Specifies a comment for the dynamic table.
```
Comment (runleonarun) [inline website/docs/reference/resource-configs/snowflake-configs.md:474]: per https://github.com/dbt-labs/dbt-snowflake/pull/1081 removing `comment`
```suggestion
```

**Review by runleonarun — CHANGES_REQUESTED** [review]
Hey @matthewshaver I noticed they have a [new PR](https://github.com/dbt-labs/dbt-snowflake/pull/1081/files) for this work. Removing `comment` per pr#1081

**Comment by HenkvanDyk** [comment]
docs added here:
https://github.com/dbt-labs/docs.getdbt.com/issues/5678

**Pull request closed without merging**